specifications/cdsc-wg1-02.md

```diff
@@ -1 +1,799 @@
+
+Severs MUST assign the submitted `redirect_uris` to all [Scope Credentials](#scope-creds-format) in the [Scope Credentials API](#scope-creds-api) that have `response_types`. Clients MAY subsequently manage which `redirect_uris` values are assigned to specific Scope Credentials using the [Scope Credentials API](#scope-creds-api).
+
+While Servers MAY support OAuth's [Dynamic Client Registration Management Protocol](https://www.rfc-editor.org/rfc/rfc7592) by including the `registration_client_uri` and `registration_access_token` values in their registration response, this specification requires that Servers MUST support the [Clients API](clients-api) by including the `cds_clients_api` value in the registration response as a way for Clients to manage their registrations.
+
+## 5. Clients API <a id="clients-api" href="#clients-api" class="permalink">🔗</a>
+
+This specification requires Servers provide a set of Application Programming Interfaces (APIs) allowing Clients to view and edit their Client registrations. These APIs are authenticated using a Bearer `access_token` granted to Clients using OAuth's [`client_credentials` grant](https://www.rfc-editor.org/rfc/rfc6749#section-4.4) and the `client_secret` provided in the initial [Client Registration Response](#registration-response) or `client_secret` values from the [Scope Credentials API](#scope-creds-api) that include the `client_admin` scope.
+
+### 5.1. Client Object Format <a id="client-format" href="#client-format" class="permalink">🔗</a>
+
+Client objects are formatted as JSON objects and contain named values. Client objects are required to follow the same object format as the [Client Registration Response](#registration-response), with the addition of the following named values.
+
+* `cds_created` - _ISO8601 datetime_ - (REQUIRED) When the Client was created.
+* `cds_modified` - _ISO8601 datetime_ - (REQUIRED) When the Client was last modified.
+* `cds_client_uri` - _URL_ - (REQUIRED) Where to submit modifications using the Clients API [Modifying Clients](#clients-modify) functionality. Servers MUST make this URL unique for each Client registration.
+
+Additionally, while the [Client Registration Response](#registration-response) MUST contain the `client_secret` field. Client objects returned from the Clients API MUST NOT include the `client_secret` or `client_secret_expires_at` fields. Clients MUST instead use the [Scope Credentials API](#scope-creds-api) to obtain `client_secret` values for use in other APIs.
```

Comment:
From @antarasargam ([on April 24, 2024](https://github.com/carbon-data-specification/Customer-Data/pull/25#discussion_r1578085034)):

---

nit: There's an abrupt period `.` in between here, probably to be replaced by a comman `,` ?

---